### PR TITLE
Brings SidebarClass a bit more line with RA, implements Zoom_Mode_Control.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: tiberiandawn/sidebar.cpp [hunk 1]
```diff
@@ -656,7 +655,8 @@ bool SidebarClass::Add(RTTIType type, int id, bool via_capture)
 bool SidebarClass::Scroll(bool up, int column)
 {
     if (Column[column].Scroll(up)) {
-        IsToRedraw = true;
+        // No need to redraw the whole sidebar juts because we scrolled a strip is there? ST - 10/15/96 7:29PM
+        // IsToRedraw = true;
```
Comment (tomsons26) [inline tiberiandawn/sidebar.cpp]: I can't run TD on my Win 10 for some reason at all, not retail or VC so need someone to confirm this doesn't change how the sidebar scrolling redraws. It should be a fluid smooth scrolling of the entire strip in TD.

**Comment by hifi** [comment]
I don't see much value in making the APIs similar without commonizing them at the same time.

**Comment by tomsons26** [comment]
> I don't see much value in making the APIs similar without commonizing them at the same time.

That would be still far away, problem mainly lies in how messy the resolution adjusting code is, i did manage to port it to RA but the result was a pretty big mess and i had no idea what is doing what.
For now i'd say this is a significant step to doing that in the future.

**Comment by tomsons26** [comment]
Removed the potential drawing changes, the sidebar draws buggy as is, ill avoid introducing more potential issues for now.

**Review by OmniBlade — APPROVED** [review]
Seems to work okay for me standalone.

**Pull request merged**